core/src/main/java/io/fabric8/maven/core/util/SpringBootUtil.java

```diff
@@ -80,25 +88,78 @@ protected static Properties getPropertiesResource(URL resource) {
     }
 
     /**
      * Returns a {@code Properties} representation of the given Yaml resource or an empty properties object if the resource is null
      */
-    protected static Properties getPropertiesFromYamlResource(URL resource) {
+    protected static Properties getPropertiesFromYamlResource(URL resource,List<String> activeProfiles) {
         if (resource != null) {
             try (InputStream yamlStream = resource.openStream()) {
-                Yaml yaml = new Yaml();
-                @SuppressWarnings("unchecked")
-                SortedMap<String, Object> source = yaml.loadAs(yamlStream, SortedMap.class);
+                Yaml yaml = new Yaml(new SafeConstructor());
+
+                Map<String, Map> profileDocs = new HashMap<>();
+
                 Properties properties = new Properties();
-                if (source != null) {
-                    properties.putAll(getFlattenedMap(source));
+
+                Iterable<Object> yamlDoc = yaml.loadAll(yamlStream);
+
+                Iterator yamlDocIterator = yamlDoc.iterator();
+
+                int docCount = 0;
+                while (yamlDocIterator.hasNext()) {
+                    Map docRoot = (Map) yamlDocIterator.next();
+
+                    String profiles = null;
+
+                    if (docRoot.containsKey("spring")) {
+
+                        LinkedHashMap value = (LinkedHashMap) docRoot.get("spring");
+
+                        Object profilesValue = value.get("profiles");
+
+                        if (profilesValue instanceof LinkedHashMap) {
+                            Map porfMap = (Map) profilesValue;
+                            String[] actProfiles = StringUtils.split((String) porfMap.get("active"), ",");
+                            if (activeProfiles.isEmpty() && docCount > 0) {
+                                activeProfiles.addAll(Arrays.asList(actProfiles));
+                            }
+                        } else if (profilesValue instanceof String) {
+                            profiles = (String) profilesValue;
+                        }
+                    }
+
+                    if (profiles != null) {
+                        String[] profileSplit = profiles.split(",");
+                        if (!CollectionUtils.
+                                intersection(Arrays.asList(profileSplit), activeProfiles)
+                                .isEmpty()) {
+                            //if the profiles is in the list of active profiles we add it to our list of docs
+                            profileDocs.put(profiles, docRoot);
+                        }
+                    } else if (docCount == 0) {
+                        //the root doc
+                        profileDocs.put("default", docRoot);
+                    }
+
+                    docCount++;
                 }
+
+                LOG.debug("Spring Boot Profile docs:{}"+profileDocs);
+
+                properties.putAll(getFlattenedMap(profileDocs.get("default")));
```

Comment:
Can't profilesDoc.get("") be null ? Thinks so, at least for certain documents. In this case an NPE will be thrown.